.buildkite/pipeline.yml

```diff
@@ -7,6 +7,7 @@ env:
   DOCKER_REGISTRY: "docker.elastic.co"
   STAGING_IMAGE: "${DOCKER_REGISTRY}/observability-ci"
   BUILDX: 1
+  GOFLAGS: "-buildvcs=false"
```

Comment:
wait, IIUC, the issue is in beats if that's the case, aren't these changes needed to be included in the consumer instead?

Reply:
For instance, the elastic-agent team did something similar and they are also consumers of this library (`golang-crossbuild`):
- https://github.com/elastic/elastic-agent/commit/cc81e4b376068eff7e15330f2c9fa0969d28c8f5

Reply:
> For instance, the elastic-agent team did something similar and they are also consumers of this library (`golang-crossbuild`):
> 
> * [elastic/elastic-agent@cc81e4b](https://github.com/elastic/elastic-agent/commit/cc81e4b376068eff7e15330f2c9fa0969d28c8f5)

FYI @v1v this PR you mentioned is interesting; the problem didn't get fixed by setting `export GOFLAGS='-buildvcs=false'` because this value gets overwritten and is hardcoded when docker is invoked here: https://github.com/elastic/beats/blob/main/dev-tools/mage/crossbuild.go#L330 ([permalink](https://github.com/elastic/beats/blob/e0675778b8eed65c6fff6a1d1911833c21563755/dev-tools/mage/crossbuild.go#L330)).

The problem got fixed by avoiding having a nested repo (i.e. cloning beats within another cloned repo (elastic-agent): https://github.com/elastic/elastic-agent/commit/cc81e4b376068eff7e15330f2c9fa0969d28c8f5#diff-f8bd7a8bc1f2826590a0bd7c0937871d04d15970ee75a830324888f63ce116f5R19 . A similar reproduction is described [here](https://github.com/golang/go/issues/49004#issuecomment-1172217880).

vcs stamping issues seem to be rampant after go 1.18.0, e.g. see https://github.com/golang/go/issues/49004 and specifically the commit description in  https://github.com/golang/go/commit/4569fe64101c2209e3429bd1c953b5f4021fc43d

cc @alexsapran